src/python/WMCore/WMRuntime/Tools/Scram.py

```diff
@@ -102,6 +102,43 @@ def isCMSSWSupported(thisCMSSW, supportedCMSSW):
 
     return False
 
+def isEnforceGUIDInFileNameSupported(thisCMSSW):
+    """
+        _isEnforceGUIDInFileNameSupported_
+
+        Function used to validate whether the CMSSW release to be used supports
+        the enforceGUIDInFileName feature.
+        :param thisCMSSW: release to be used in this job
+
+    """
+    # a set of CMSSW releases that support the enforceGUIDInFileName feature. Releases in the same
+    # cycle with a higher minor revision number also support the feature.
+    supportedReleases = set(["CMSSW_10_6_8", "CMSSW_10_2_20", "CMSSW_9_4_16", "CMSSW_9_3_17", "CMSSW_8_0_34"])
+    # a set of specific CMSSW releases that supported the enforceGUIDInFileName feature.
+    specificSupportedReleases = set(["CMSSW_10_2_20_UL", "CMSSW_9_4_16_UL", "CMSSW_8_0_34_UL", "CMSSW_7_1_45_patch3"])
+
+    if not thisCMSSW:
```

Comment:
Is it because it could be an empty string or another value that evaluates to False (like None)?

Reply:
There is a (unlikely) chance that `getCmsswVersion()` would return an empty string. 
https://github.com/dmwm/WMCore/blob/f4883a8de9e88fe7fba8ea1913308d20f4524560/src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py#L650